### DISCUSSION TITLE
[x][closed][trackfind]Study->Publications is currently an array, but TrackFind does not handle arrays well

### DISCUSSION THREAD

**Discussion by sveinugu** [fairtracks/fairtracks_standard#115] [Team Posts]
..except as defining the objects of a top-level category. Some options:

1. The TrackFind way would thus be to extract the publications into a separate top-level object type. We could in this case for instance automatically add the paper title and authors in separate fields, but I fear we are overdoing it. 

2. Alternatively, we can only accept a single publication.

3. Or, we can keep as is, and
a. try to implement array handling in TrackFind search (which would probably be very complicated and confusing)
b. ignore the issue and hope no-one will search using publication ids...

**Comment by dzerbino** [comment]
Let's look at the long and short:
- short term: there is only one trackhub for the prototype and we control it, so we can agree that there is only one publication per study
- long term: I don't think we should set the standard around the implementation details of TF.

**Comment by sveinugu** [comment]
@dtitov I have an idea now of how we could implement array search in TrackFind. One could extract all arrays as separate object types internally when crawling, and then reinsert them as arrays in the search results, or for validation, keeping the standard and the implementation separate. Also one could make this invisible to normal users (but probably not for curators). This would be an extension for after the deadline, if we are able to keep the standard without nested arrays for now.

**Comment by sveinugu** [comment]
We could have a similar solution for the id->object maps that can be found in some jsons (e.g. IHEC), which is also not directly supported by TrackFind (it does not support searching for attributes labels [the specific ids in this case], only leaf values). Automatic transformation into object types when crawling (might even be possible to auto detect such structures), and transformation back in search results and validation.

**Comment by sveinugu** [comment]
Added an issue to remove the array short-term: https://github.com/fairtracks/fairtracks_standard/issues/13

**Comment by sveinugu** [comment]
Added an issue on the possible array search support in TrackFind: https://github.com/elixir-no-nels/trackfind/issues/97. Closing this
